packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -0,0 +1,355 @@
+            updateReferencedDataStores(dataStore.id);
+        }
+    };
+
+    const unreferenceDataStore = (dataStore: TestDataObject, parentDataStore: TestDataObject) => {
+        if (rootDataStores.has(dataStore.id)) {
+            return;
+        }
+        parentDataStore._root.delete(dataStore.id);
+        getTestNode(parentDataStore.id).children.delete(dataStore.id);
+        getTestNode(dataStore.id).parents.delete(parentDataStore.id);
+        updateUnreferencedDataStores(dataStore.id);
+    };
+
+    const getRandomContainer = (): IContainer => {
+        const id = randomInteger(0, connectedContainers.length - 1);
+        return connectedContainers[id];
+    };
+
+    const getRandomDataStoreFromContainer = async (container: IContainer) => {
+        const random = randomInteger(0, testNodes.size - 1);
+        const dataStoreId = Array.from(testNodes.values())[random].id;
+        return requestFluidObject<TestDataObject>(container, dataStoreId);
+    };
+
+    const getRandomParentForDataStore = async (dataStore: TestDataObject, container: IContainer) => {
+        const parents = getTestNode(dataStore.id).parents;
+        const random = randomInteger(0, parents.size - 1);
+        const parentId = Array.from(parents.values())[random];
+        return requestFluidObject<TestDataObject>(container, parentId);
+    };
+
+    const referenceRandomDataStore = async () => {
+        if (connectedContainers.length <= 0 || testNodes.size <= 0) {
+            return;
+        }
+        const container = getRandomContainer();
+        const dataStore = await getRandomDataStoreFromContainer(container);
+        const parent = await getRandomDataStoreFromContainer(container);
+        referenceDataStore(dataStore, parent);
+    };
+
+    const unreferenceRandomDataStore = async () => {
+        if (connectedContainers.length <= 0 || testNodes.size <= 0) {
+            return;
+        }
+        const container = getRandomContainer();
+        const dataStore = await getRandomDataStoreFromContainer(container);
+        const parentDataStore = await getRandomParentForDataStore(dataStore, container);
+        unreferenceDataStore(dataStore, parentDataStore);
+    };
+
+    const actionsList: (() => Promise<unknown>)[] = [
+        loadNewContainer,
+        closeRandomContainer,
+        createDataStoreForRandomContainer,
+        referenceRandomDataStore,
+        unreferenceRandomDataStore,
+    ];
+    let testTime = 0;
+    const testEnd = 60 * 1000; // 1 minute
+    const testActions: { sleep: number; action: () => Promise<unknown>; }[] = [];
+    while (testTime < testEnd) {
+        const sleepTime: number = randomInteger(0, defaultSessionExpiryDurationMs + 1000);
+        const action = makeRandom().pick(actionsList);
+        testActions.push({ sleep: sleepTime, action });
+        testTime += sleepTime;
+    }
+
+    it("Create and reference and unreference datastores with multiple containers", async () => {
```

Comment:
We should think about repeatability/observability here. If a test run exposes a bug... will you be able to find out the sequence of events that caused it?

You can use the seed argument for `makeRandom` to ensure repeatability.  Even if the seed itself is random, you can at least log it (include it in the test title, perhaps).  Or you can lock the seed in and then it's deterministic, you just didn't have to write out the script, it was generated for you.

You could also consider printing out the whole `testActions` object if there's an error.